### PR TITLE
Add Javascript game class

### PR DESCRIPTION
**Pull request by Khufu-I**
Encapsulate Tic Tac Toe game logic into it's own class for easier testing and simulation.

### CODE REVIEW

File: src/lib/Game.ts [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+function isWinPattern(toes: string[], winPattern: number[]) {
+  return (
+    toes[winPattern[0]] === toes[winPattern[1]] &&
+    toes[winPattern[1]] === toes[winPattern[2]] &&
+    toes[winPattern[0]] !== ''
+  )
+}
+
+export default class Game{
+  #isXTurn: boolean;
```
Comment (Vithop) [inline src/lib/Game.ts:48]: we don't need to use the ugly # for private members in typescript we can use private. https://www.typescriptlang.org/docs/handbook/2/classes.html#private

File: src/lib/Game.ts [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    this.#isXTurn = isXTurn;
+    this.#boardState = boardState;
+  }
+
+  get isXTurn(): boolean { return this.#isXTurn; }
+  get boardState(): string[] { return this.#boardState; }
+  get status() {
+    const winPositions = winMap.find(winPattern => isWinPattern(this.#boardState, winPattern));
+    if(winPositions){
+      switch(this.#boardState[winPositions.at(0)]){
```
Comment (Vithop) [inline src/lib/Game.ts:60]: Why do we need this switch case when we already have who's turn it is tracked? Can't we do something like `Status[isXTurn ? 'X' : 'O'].won()`

File: src/lib/Game.ts [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+ * @example
+ *   const game = new Game();
+ *   game.executeMove(3);
+ */
+
+export const Status = {
+  XTurn: 0,
+  OTurn: 1,
+  Draw: 2,
+  XWon: class {
```
Comment (Vithop) [inline src/lib/Game.ts:14]: Why is this a class it does so little?

File: src/routes/TicTacToe/+page.svelte [hunk 4]
```diff
@@ -107,12 +58,9 @@
 
 <svelte:window on:load={foo} />
 
-<!-- <h1>Tic Tac Toe</h1>
-<h1>Mathu is {vithu}</h1>
```
Comment (Vithop) [inline src/routes/TicTacToe/+page.svelte:111]: 🙀😱😮

File: src/lib/Game.ts (hunk 1 above)
Comment (Khufu-I) [reply src/lib/Game.ts:48]: [Actually Typescript private doesn’t do anything at runtime](https://www.typescriptlang.org/docs/handbook/2/classes.html#caveats) and JS private fields should be preferred

File: src/lib/Game.ts (hunk 2 above)
Comment (Khufu-I) [reply src/lib/Game.ts:60]: The way this class is currently setup you can construct a Game where isXTurn is true but actually O won.

I’m thinking about whether to store isXTurn or Status but that will be another diff.

File: src/lib/Game.ts (hunk 3 above)
Comment (Khufu-I) [reply src/lib/Game.ts:14]: It's essentially a Status enum but it holds the winning positions. Not sure if there's a better way to do this in Typescript.

Comment (Vithop) [reply src/lib/Game.ts:14]: We could just use a normal object it doesn't need a constructor

**Review by Vithop — APPROVED** [review]

**Pull request merged**